File: src/SOP8/Base/Account/Base.Account.IBLL/Request/RequestLoginKey.cs
namespace Base.Account.IBLL.Request
{
    public class RequestLoginKey
    {
        public enum ModeType { Email = 0, PhoneNumber };

        private long? num = null;
        private string m_strUserID = null;
        private string m_strName = null;
        private string m_strData = null;
        private int? m_nMode = null;

        public long? Num
        {
            get { return num; }
            set { num = value; }
        }

        public string UserID
        {
            get { return m_strUserID; }
            set { m_strUserID = value; }
        }

        public string Name
        {
            get { return m_strName; }
            set { m_strName = value; }
        }

        public string Data
        {
            get { return m_strData; }
            set { m_strData = value; }
        }

        // ModeType
        public int? Mode
        {
            get { return m_nMode; }
            set { m_nMode = value; }
        }
    }
}

File: src/SOP8/광양/EgovLogin/base-account-ibll/src/main/java/egovframework/base/account/ibll/request/RequestLoginKey.java
package egovframework.base.account.ibll.request;

public class RequestLoginKey {

    public enum ModeType {
        Email(0),
        PhoneNumber(1);

        private final int code;

        ModeType(int code) {
            this.code = code;
        }

        public int getCode() {
            return code;
        }
    }

    private Long num = null;
    private String userID = null;
    private String name = null;
    private String data = null;
    private Integer mode = null;

    public Long getNum() {
        return num;
    }

    public void setNum(Long num) {
        this.num = num;
    }

    public String getUserID() {
        return userID;
    }

    public void setUserID(String userID) {
        this.userID = userID;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getData() {
        return data;
    }

    public void setData(String data) {
        this.data = data;
    }

    public Integer getMode() {
        return mode;
    }

    public void setMode(Integer mode) {
        this.mode = mode;
    }
}